setup.py

```diff
@@ -658,27 +672,46 @@ def _read_requirements(filename: str) -> list[str]:
         import platform
 
         arch = platform.machine()
-        if arch == "x86_64":
-            wheel_tag = "manylinux1_x86_64"
-        elif arch == "aarch64":
-            wheel_tag = "manylinux2014_aarch64"
-        else:
-            raise ValueError(f"Unsupported architecture: {arch}")
-        base_commit = precompiled_wheel_utils.get_base_commit_in_main_branch()
-        wheel_url = f"https://wheels.vllm.ai/{base_commit}/vllm-1.0.0.dev-cp38-abi3-{wheel_tag}.whl"
-        nightly_wheel_url = (
-            f"https://wheels.vllm.ai/nightly/vllm-1.0.0.dev-cp38-abi3-{wheel_tag}.whl"
-        )
-        from urllib.request import urlopen
-
+        # try to fetch the wheel metadata from the nightly wheel repo
+        variant = "cu" + envs.VLLM_MAIN_CUDA_VERSION.replace(".", "")
```

Comment:
we can introduce a new env var like `VLLM_PRECOMPILED_VARIANT` to explicitly set to `cu129`/`cu130`, or potentially extended for `cpu` in the future (and empty by default). no need to be hard-coded and coupled with `VLLM_MAIN_CUDA_VERSION`